src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1027,7 +1023,7 @@ internal static bool IsValidV6SwitchGoverningType(this TypeSymbol type, bool isT
             // User-defined implicit conversion with target type as Enum type is not valid.
             if (!isTargetTypeOfUserDefinedOp && type.IsEnumType())
             {
-                type = type.GetEnumUnderlyingType();
+                type = type.GetEnumUnderlyingType()!;
```

Comment:
The call to `IsEnumType` in the condition can be removed, and this line simplified:

```suggestion
                type = type.GetEnumUnderlyingTypeOrSelf();
```